### ISSUE TITLE
Embedding backward gradient/index shape mismatch

### ISSUE THREAD

**Issue by ndrakulicTT** [tenstorrent/tt-xla#3519]
**Affects:** gpt_oss/pytorch-20B

**Error:**
```
TT_FATAL @ embedding_backward_device_operation.cpp:73:
  grad_tensor_shape[2] == index_tensor_shape[0] * index_tensor_shape[-1]
info: Number of rows in gradient tensor must be equal to number of indices in index tensor
```

**Repro:** If PR #3092 is not yet merged, run on branch `ndrakulic/torch_mc_training`:
```bash
pytest -svv tests/runner/test_models.py::test_all_models_torch[gpt_oss/pytorch-20B-tensor_parallel-training]
```
